R/SEIRDV.R

```diff
@@ -0,0 +1,364 @@
+        init_cond$R0 + init_cond$V0 != 1) {
+      stop("Invalid initial conditions. Must add up to 1.")
+    }
+    
+    object@initial_conditions <- init_cond
+    
+    object
+  })
+
+#' @describeIn SEIRDV Set transmission parameters (beta, kappa, gamma, mu, nu,
+#' delta_V, delta_R) of the SEIRV model.
+#'
+#' If the transmission parameters provided to are not 1-dimensional an error is
+#' thrown.
+#'
+#' @param object (SEIRDV model)
+#' @param value (list) list of values for beta, kappa, gamma, mu, nu, delta_V and
+#' delta_R respectively.
+#'
+#' @return object of class SEIRDV with transmission parameter values
+#' assigned.
+#' 
+#' @export
+setMethod(
+  "transmission_parameters<-", "SEIRDV",
+  function(object, value) {
+    
+    # create list of parameter values
+    if (mean(names(value) %in% object@transmission_parameter_names) != 1)
+      stop(paste0("Transmission parameters must contain: ",
+                  object@transmission_parameter_names))
+    trans_params <- value
+    
+    # check format of parameters
+    if (length(trans_params$beta) != 1
+        | length(trans_params$kappa) != 1
+        | length(trans_params$gamma) != 1
+        | length(trans_params$mu) != 1
+        | length(trans_params$nu) != 1
+        | length(trans_params$delta_V) != 1
+        | length(trans_params$delta_R) != 1) {
+      stop("The parameter values should be 1-dimensional.")
+    }
+    
+    # if all above tests are passed, assign the trans_params namelist to the
+    # object
+    object@transmission_parameters <- trans_params
+    
+    object
+  })
+
+# SEIRDV class specific functions
+
+#' Retrieves intervention parameters of SEIRDV model.
+#'
+#' @param object An object of the class SEIRDV.
+#' 
+#' @export
+setGeneric("intervention_parameters",
+           function(object) standardGeneric("intervention_parameters"))
+
+
+#' @describeIn SEIRDV Retrieves intervention parameters of SEIRDV model.
+#'
+#' @param object An object of the class SEIRDV.
+#' 
+#' @export
+setMethod("intervention_parameters", "SEIRDV",
+          function(object) object@intervention_parameters)
+
+#' Set intervention parameters of the SEIRV model.
+#'
+#' Intervention parameters have same size. This class is designed for interventions
+#' which last several days at least and have several days between them.
+#'
+#' @param object an object of the class SEIRDV
+#' @param value (list) list of intervention parameters: starts, stops and
+#'              coverages.
+#'
+#' @return object of class SEIRDV with intervention parameters assigned.
+#' 
+#' @export
+setGeneric(
+  "intervention_parameters<-",
+  function(object, value) {
+    standardGeneric("intervention_parameters<-")
+  })
+
+#' @describeIn SEIRDV Setter method for intervention parameters of the SEIRV model.
+#'
+#' Intervention parameters have same size. This class is designed for interventions
+#' which last several days at least and have several days between them.
```

Comment:
```suggestion
#' Intervention parameters have same size. A tanh function is used to smooth interventions during simulation. This class is designed for interventions
#' which last several days at least and have several days between them; interventions involving rapid fluctuations may be distorted.
```
Just in case anyone needs to know why